File: C#/PokerWorld/Data/IDataPersistance.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace PokerWorld.Data
{
    public interface IDataPersistance
    {
        bool IsUsernameExist(string username);
        bool IsDisplayNameExist(string displayName);
        void Register(UserInfo u);
        UserInfo Get(string username);
        UserInfo Authenticate(string username, string password);
        void Update(UserInfo u);
        void Delete(UserInfo u);
    }
}

File: trunk/JAVA/PokerWorld/src/bluffinmuffin/data/IDataPersistance.java
package bluffinmuffin.data;

public interface IDataPersistance
{
    boolean isUsernameExist(String username);
    
    boolean isDisplayNameExist(String displayName);
    
    void register(UserInfo u);
    
    UserInfo get(String username);
    
    UserInfo authenticate(String username, String password);
    
    void update(UserInfo u);
    
    void delete(UserInfo u);
}
